executor/write.go

```diff
@@ -912,7 +912,7 @@ func (e *InsertValues) initDefaultValues(row []types.Datum, marked map[int]struc
 				return errors.Trace(err)
 			}
 			row[i].SetInt64(val)
-			if val != 0 {
+			if val != 0 || (e.ctx.GetSessionVars().SQLMode&mysql.ModeNoAutoValueOnZero) > 0 {
```

Comment:
In MySQL version: 5.7.15 the behavior is followings:
```
mysql> show create table t;
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                     |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` int(11) NOT NULL AUTO_INCREMENT,
  `b` int(11) DEFAULT NULL,
  PRIMARY KEY (`a`)
) ENGINE=InnoDB AUTO_INCREMENT=11 DEFAULT CHARSET=utf8 |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
mysql> insert into t value(0,0);
Query OK, 1 row affected (0.01 sec)
mysql> insert into t value(0,0);
Query OK, 1 row affected (0.01 sec)
mysql> set sql_mode=NO_AUTO_VALUE_ON_ZERO;
Query OK, 0 rows affected, 1 warning (0.00 sec)
mysql> insert into t value(0,0);
ERROR 1062 (23000): Duplicate entry '0' for key 'PRIMARY'
```
You can try on MySQL 5.6.